solr/core/src/java/org/apache/solr/request/SimpleFacets.java

```diff
@@ -325,6 +329,9 @@ public void getFacetQueryCount(ParsedParams parsed, NamedList<Integer> res) thro
    * @see FacetParams#FACET_QUERY
    */
   public int getGroupedFacetQueryCount(Query facetQuery, DocSet docSet) throws IOException {
+    if (docSet.size() == 0) {
+      return 0;
+    }
```

Comment:
Both callers of this method (`SimpleFacets.getFacetQueryCount(...)` and `RangeFacetProcessor.rangeCount(...)` check for the base `docSet.size()==0` case independently. Could perhaps remove this redundant check (or replace with `assert docSet.size() > 0`)?

Reply:
Same as above